connector/src/main/java/org/jboss/as/connector/deployers/datasource/DefaultDataSourceResourceReferenceProcessor.java

```diff
@@ -48,7 +49,9 @@ public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitPro
         if(deploymentUnit.getParent() == null) {
             final EEResourceReferenceProcessorRegistry eeResourceReferenceProcessorRegistry = deploymentUnit.getAttachment(Attachments.RESOURCE_REFERENCE_PROCESSOR_REGISTRY);
             if(eeResourceReferenceProcessorRegistry != null) {
-                eeResourceReferenceProcessorRegistry.registerResourceReferenceProcessor(RESOURCE_REFERENCE_PROCESSOR);
+                final EEModuleDescription eeModuleDescription = deploymentUnit.getAttachment(Attachments.EE_MODULE_DESCRIPTION);
+                if(eeModuleDescription != null && eeModuleDescription.getDefaultResourceJndiNames().getDataSource() != null)
```

Comment:
This is always `false` because `DefaultDataSourceResourceReferenceProcessor` (Phase.STRUCTURE) gets executed before `DefaultBindingsConfigurationProcessor` (Phase.PARSE), the latter set the default datasource if any.

This will cause that `@Resource() DataSource ds` won't get injected.

Reply:
I suggest we look into getting rid of DefaultBindingsConfigurationProcessor and collapse its logic into EEModuleInitialProcessor. When EEModuleInitialProcessor creates and attaches the EEModuleDescription all the default resource jndi names are known, as they all come from the subsystem config. There's no new information that comes in between then and Phase.PARSE when DefaultBindingsConfigurationProcessor runs. 

The approach @bmaxwell proposes here would then work. But it should be applied to the other analogous DUPs as well, which I believe will have the same problem.